src/base/BaseActionsRouter.sol

```diff
@@ -16,12 +16,22 @@ abstract contract BaseActionsRouter is SafeCallback {
     /// @notice emitted when an inheriting contract does not support an action
     error UnsupportedAction(uint256 action);
 
+    error Wrap__SubcontextReverted(address subcontext, bytes reason);
+    error Wrap__CurrencyNotSettled(address subcontext, bytes reason);
+
     constructor(IPoolManager _poolManager) SafeCallback(_poolManager) {}
 
     /// @notice internal function that triggers the execution of a set of actions on v4
     /// @dev inheriting contracts should call this function to trigger execution
     function _executeActions(bytes calldata unlockData) internal {
-        poolManager.unlock(unlockData);
+        try poolManager.unlock(unlockData) {}
+        catch (bytes memory error) {
```

Comment:
I would suggest renaming `error` because it's a reserved keyword

Reply:
we usually use `reason`... and it will also match the input in the bubble up error definitions